file.go

```diff
@@ -39,47 +48,52 @@ func (t *TOMLLoader) Load(s interface{}) error {
 }
 
 // JSONLoader satisifies the loader interface. It loads the configuration from
-// the given json file.
+// the given json file or Reader
 type JSONLoader struct {
-	Path string
+	Path   string
+	Reader io.Reader
 }
 
 // Load loads the source into the config defined by struct s
+// Defaults to using the Reader if provided, otherwise tries to read from the
+// file
 func (j *JSONLoader) Load(s interface{}) error {
-	filePath, err := getConfigPath(j.Path)
-	if err != nil {
-		return err
-	}
-
-	file, err := ioutil.ReadFile(filePath)
-	if err != nil {
-		return err
+	var r io.Reader
+	if j.Reader != nil {
+		r = j.Reader
+	} else {
```

Comment:
Just like in the TOML loader, can you return an error if both fields are not initialized?